src/hooks/App.js

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react'
+
+import '../shared/App.css'
+import { Timeline } from './Timeline'
+import { Video } from './Video'
+import { VideoPlayer } from './VideoPlayer'
+
+const baseClass = 'app'
+const HIDE_MINI_PLAYER = 'HIDE_MINI_PLAYER'
+const SHOW_MINI_PLAYER = 'SHOW_MINI_PLAYER'
+const initialState = {
+  showMiniPlayer: false,
+  startAt: 0,
+  left: 0,
+}
+
+const reducer = (state, action) => {
+  const { payload, type } = action
+  console.log('dispatching', type)
+
+  switch (type) {
+    case SHOW_MINI_PLAYER:
+      return {
+        ...state,
+        left: payload.left,
+        showMiniPlayer: true,
+        startAt: payload.startAt,
+      }
+    case HIDE_MINI_PLAYER:
+      return initialState
+    default:
+      return state
+  }
+}
+
+export const App = () => {
+  const [state, dispatch] = React.useReducer(reducer, initialState)
+  const hideMiniPlayer = () => dispatch({ type: HIDE_MINI_PLAYER })
+  const showMiniPlayer = (startAt, left) =>
```

Comment:
Can never tell when arguments should be positional vs named

Reply:
yeah this was just yoinked from the previous code but I agree it should be named. Not going to do it now because I want to solve the immediate problem but can go back and refactor later. I'm sure there are a ton of refactor opportunities in this app.